midburn_queue.rb

```diff
@@ -21,40 +28,48 @@ class MidburnQueue < Sinatra::Base
   end
 
   before :method => :post do
-    # http://stackoverflow.com/questions/15671006/before-filter-for-all-post-requests-in-sinatra
-    response.headers["Access-Control-Allow-Headers"] = "X-Requested-With, X-HTTP-Method-Override, Content-Type, Cache-Control, Accept"
-    response.headers["Access-Control-Allow-Origin"] = ENV["ACCESS_CONTROL_ALLOW_ORIGIN"]
+    load_params
+    set_response_headers
+    access_log
   end 
 
-  def get_params
+  def access_log
+    puts "[access log] #{self.env["REQUEST_METHOD"]} #{self.env["REQUEST_PATH"]} from #{self.env["REMOTE_ADDR"]}: #{params}"
+  end
+
+  def set_response_headers
+    response.headers["Access-Control-Allow-Headers"] = "X-Requested-With, X-HTTP-Method-Override, Content-Type, Cache-Control, Accept"
+    response.headers["Access-Control-Allow-Origin"] = ACCESS_CONTROL_ALLOW_ORIGIN
+  end
+
+  def load_params
+    return @params if @request_body_read
     begin
-      JSON.parse(request.body.read)
+      @params = JSON.parse(request.body.read)
```

Comment:
Good point, I like OJ, but for such a small payload (i.e. `{"email":"user@example.com"}`), I don't think we will gain a lot by switching.